stdlib/gc.ml

```diff
@@ -165,4 +168,6 @@ module Memprof =
       set_ctrl sampling_rate callstack_size minor_alloc_callback
                major_alloc_callback promote_callback minor_dealloc_callback
                major_dealloc_callback
+
+    let () = at_exit (fun () -> if !has_started then stop ())
```

Comment:
Can you confirm that unconditionally calling `stop ()` here would be correct (looking at the code of `caml_memprof_set`, I assume that it is an indempotent call that is a no-op if `start ()` was not called before), but the boolean flag is here for performance purposes? (No noticeable computational cost unless Memprof was used.)

I would find it slightly more natural to have a way in the API to ask if memprof is currently sampling (I guess this mean checking `sampling_rate > 0.`), and to stop only in this case. In terms of performance, this is not strictly equivalent, given that your approach will stop even if the program started and then stopped manually.

(Also the implementation would not rely on an extra piece of global mutable state, using only the memprof state.)